app/controllers/recipes_controller.rb

```diff
@@ -1,4 +1,5 @@
 class RecipesController < ApplicationController
+  before_action :authenticate_user!, except: :show
```

Comment:
Excellent use of `except:` here, instead of `only:`, well done. 

Additional actions added to `RecipesController` in the future will be more secure by default thanks to that choice.